components/fxa-client/src/devices.rs

```diff
@@ -0,0 +1,474 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+//! # Device Management
+//!
+//! Applications that connect to a user's account may register additional information
+//! about themselves via a "device record", which allows them to:
+//!
+//!    - customize how they appear in the user's account management page
+//!    - receive push notifications about events that happen on the account
+//!    - participate in the FxA "device commands" ecosystem
+//!
+//! For more details on FxA device registration and management, consult the
+//! [Firefox Accounts Device Registration docs](
+//! https://github.com/mozilla/fxa/blob/main/packages/fxa-auth-server/docs/device_registration.md).
```

Comment:
I wonder if this could be split up more, maybe `devices.rs` and `push.rs`?  WDYT?

Reply:
That sounds reasonable, maybe we should go further and consolidate the modules inside the `internal` module with the modules at this layer?